### PR TITLE
[process-agent] Remove pkg/process/config package

### PR DESCRIPTION
**Pull request by xornivore**
### What does this PR do?

- Removes pkg/process/config package
- Move config bootstrap helpers into cmd/process-agent/command (for now)
- Provides env var bindings for legacy orchestrator settings
- Adds tests to pkg/orchestrator/config for legacy orchestrator settings

### Motivation

Remove pkg/process/config in preparation for using components.

### Describe how to test/QA your changes

Validate functionality related to loading configuration files and env vars in the process agent.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by JLineaweaver — APPROVED** [review]
Looks like a good cleanup to me.

**Review by just-chillin — APPROVED** [review]
Great refactor!

File: cmd/process-agent/command/command.go [hunk 1]
```diff
@@ -106,18 +110,89 @@ func MakeCommand(subcommandFactories []SubcommandFactory, winParams bool, rootCm
...
+func setHostMountEnv() {
+	// Set default values for proc/sys paths if unset.
+	// Don't set this is /host is not mounted to use context within container.
+	// Generally only applicable for container-only cases like Fargate.
+	if !config.IsContainerized() {
+		return
+	}
+
+	_, err := os.Stat("/host")
+	if !os.IsNotExist(err) {
```
Comment (kkhor-datadog) [inline cmd/process-agent/command/command.go:165]: Not sure how much of an impact this has, but it seems like there is a small change in behavior - previously we were using `util.PathExists`, which return true only if there was no error returned from `os.Stat`, whereas here we continue if `!os.IsNotExist`. In particular it seems like if the error was something like a permission error we'd see different behavior compared to the previous implementation which would not set these environment variables, whereas this one would

Comment (xornivore) [reply cmd/process-agent/command/command.go:165]: Yep, thank you, nice catch! To think of it, the helper exists in the util which will still be around for some time, no reason not to use it.

Comment (kkhor-datadog) [reply cmd/process-agent/command/command.go:165]: Sounds good, looks like there is another occurrence of this in `loadConfigIfExists` below as well

**Review by hmahmood — APPROVED** [review]

**Review by kkhor-datadog — APPROVED** [review]

**Review by sgnn7 — APPROVED** [review]
LGTM with a few nits though it would be good to have @ogaca-dd or @truthbk take a look as well.

File: cmd/process-agent/command/command.go [hunk 2]
```diff
@@ -106,18 +111,82 @@ func MakeCommand(subcommandFactories []SubcommandFactory, winParams bool, rootCm
...
+	if err := loadConfigIfExists(path); err != nil {
 		return err
 	}
 
+	// Resolve any secrets
+	if err := config.ResolveSecrets(config.Datadog, filepath.Base(path)); err != nil {
+		return err
+	}
+
+	var (
```
[the remark this answers is not in the document]
Comment (xornivore) [reply cmd/process-agent/command/command.go:126]: I kept them closer to where they were initialized to the config as opposed to the beginning of the func where we have a couple of early exit cases. 

Regarding the config defaults, while the defaults do make sense for the daemon execution, for one shot commands they may not.

**Pull request merged**